esp/logging/logginglib/logthread.cpp

```diff
@@ -414,3 +400,29 @@ IEspUpdateLogRequestWrap* CLogThread::unserializeLogRequestContent(const char* l
 
     return new CUpdateLogRequestWrap(guid, opt, buffer.str());
 };
+
+void CLogThread::writeJobQueue(IEspUpdateLogRequestWrap* jobToWrite)
+{
+	if (jobToWrite)
+	{
+		CriticalBlock b(logQueueCrit);
+		int QueueSize = logQueue.ordinality();
+		if(QueueSize > maxLogQueueLength)
+			ERRLOG("LOGGING QUEUE SIZE %d EXECEEDED MaxLogQueueLength %d, check the logging server.",QueueSize, maxLogQueueLength);
+
+		if(QueueSize!=0 && QueueSize % signalGrowingQueueAt == 0)
+			ERRLOG("Logging Queue at %d records. Check the logging server.",QueueSize);
+
+		logQueue.enqueue(LINK(jobToWrite));
+	}
+}
+
+IEspUpdateLogRequestWrap* CLogThread::readJobQueue()
+{
+	CriticalBlock b(logQueueCrit);
+	if (logQueue.ordinality() > 0)
```

Comment:
minor: This test is redundant since dequeue() already returns null if it is empty.